### PR TITLE
P2P - tenant <--> protocol mapping

### PR DESCRIPTION
**Pull request by vimukthi-git**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: #616

### CODE REVIEW

File: p2p/client.go [hunk 1]
```diff
@@ -102,18 +106,18 @@ func (s *peer) getPeerID(id identity.Identity) (libp2pPeer.ID, error) {
 
 // getSignatureForDocument requests the target node to sign the document
 func (s *peer) getSignatureForDocument(ctx context.Context, identityService identity.Service, doc coredocumentpb.CoreDocument, receiverPeer libp2pPeer.ID, receiverCentID identity.CentID) (*p2ppb.SignatureResponse, error) {
-	senderID, err := s.config.GetIdentityID()
+	self, err := contextutil.Self(ctx)
```
Comment (mikiquantum) [inline p2p/client.go]: I was making these changes as well, thanks.

File: p2p/receiver/protocol.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package receiver
```
Comment (mikiquantum) [inline p2p/receiver/protocol.go]: I created a different package in my pr for some common sender/receiver utils, I will move that there as part of my work.

File: p2p/server.go [hunk 3]
```diff
@@ -71,39 +60,60 @@ func (*peer) Name() string {
 func (s *peer) Start(ctx context.Context, wg *sync.WaitGroup, startupErr chan<- error) {
 	defer wg.Done()
 
-	if s.config.GetP2PPort() == 0 {
+	nc, err := s.config.GetConfig()
```
Comment (mikiquantum) [inline p2p/server.go]: Cool

File: p2p/server.go [hunk 4]
```diff
@@ -71,39 +60,60 @@ func (*peer) Name() string {
...
+	var protocols []protocol.ID
+	for _, t := range tcs {
+		CID, err := identity.ToCentID(t.IdentityID)
+		if err != nil {
+			startupErr <- err
+			return
+		}
+		protocols = append(protocols, receiver.ProtocolForCID(CID))
+	}
+	s.mes.init(protocols...)
```
Comment (mikiquantum) [inline p2p/server.go]: We briefly talked about adding the protocol list to the messenger struct and let the init iterate over them? I am ok leaving it as is. The only question for me is if we see a use case on having them as part of the struct.

Comment (vimukthi-git) [reply p2p/server.go]: Thats part of the messenger isn't it? so there won't be a change here even if we implement your suggested change.

**Comment by vimukthi-git** [comment]
@mikiquantum I don't know. It seems there were some changes.

File: p2p/server.go (hunk 4 above)
Comment (mikiquantum) [reply p2p/server.go]: If we have the protocols within the messenger then we do not need to pass protocol list to the `init` method. Just `s.mes.init()`.

**Pull request merged**